examples/consumer/main.go

```diff
@@ -178,7 +207,14 @@ RETRY:
 			if !healthOK {
 				deleteAllSubscriptions()
 				log.Info("delete all subscriptions due to publisher health failure")
-				return
+				goto RETRY
```

Comment:
When we retry, is there a theoretical possibility of multiple retries? If yes, could it be beneficial to introduce a sleep before retrying?

Reply:
deleteAllSubscriptions() will set the `subOK` to false, so retry will go to different path. If the new path failed it will retry after 30 seconds.